src/code/z_parameter.c

```diff
@@ -52,38 +332,750 @@
+        }
+        return ITEM_NONE;
+
+    } else if (item == ITEM_HEART_CONTAINER) {
+        gSaveContext.save.playerData.healthCapacity += 0x10;
+        gSaveContext.save.playerData.health += 0x10;
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SONG_SONATA) && (item <= ITEM_SONG_LULLABY_INTRO)) {
+        SET_QUEST_ITEM(item - ITEM_SONG_SONATA + QUEST_SONG_SONATA);
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SWORD_KOKIRI) && (item <= ITEM_SWORD_GILDED)) {
+        SET_EQUIP_VALUE(EQUIP_SWORD, item - ITEM_SWORD_KOKIRI + 1);
+        BUTTON_ITEM_EQUIP(CUR_FORM, EQUIP_SLOT_B) = item;
+        Interface_LoadItemIconImpl(globalCtx, 0);
+        if (item == ITEM_SWORD_RAZOR) {
+            gSaveContext.save.playerData.swordHealth = 100;
+        }
+        return ITEM_NONE;
+
+    } else if ((item >= ITEM_SHIELD_HERO) && (item <= ITEM_SHIELD_MIRROR)) {
+        if (GET_CUR_EQUIP_VALUE(EQUIP_SHIELD) != (u16)(item - ITEM_SHIELD_HERO + 1)) {
+            SET_EQUIP_VALUE(EQUIP_SHIELD, item - ITEM_SHIELD_HERO + 1);
+            Player_SetEquipmentData(globalCtx, player);
+            return ITEM_NONE;
+        }
+        return item;
+
+    } else if ((item == ITEM_KEY_BOSS) || (item == ITEM_COMPASS) || (item == ITEM_DUNGEON_MAP)) {
+        SET_DUNGEON_ITEM(item - ITEM_KEY_BOSS, gSaveContext.mapIndex);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_KEY_SMALL) {
+        if (DUNGEON_KEY_COUNT(gSaveContext.mapIndex) < 0) {
+            DUNGEON_KEY_COUNT(gSaveContext.mapIndex) = 1;
+            return ITEM_NONE;
+        } else {
+            DUNGEON_KEY_COUNT(gSaveContext.mapIndex)++;
+            return ITEM_NONE;
+        }
+
+    } else if ((item == ITEM_QUIVER_30) || (item == ITEM_BOW)) {
+        if (CUR_UPG_VALUE(UPG_QUIVER) == 0) {
+            Inventory_ChangeUpgrade(UPG_QUIVER, 1);
+            INV_CONTENT(ITEM_BOW) = ITEM_BOW;
+            AMMO(ITEM_BOW) = CAPACITY(UPG_QUIVER, 1);
+            return ITEM_NONE;
+        } else {
+            AMMO(ITEM_BOW)++;
+            if (AMMO(ITEM_BOW) > (s8)CUR_CAPACITY(UPG_QUIVER)) {
+                AMMO(ITEM_BOW) = CUR_CAPACITY(UPG_QUIVER);
+            }
+        }
+    } else if (item == ITEM_QUIVER_40) {
+        Inventory_ChangeUpgrade(UPG_QUIVER, 2);
+        INV_CONTENT(ITEM_BOW) = ITEM_BOW;
+        AMMO(ITEM_BOW) = CAPACITY(UPG_QUIVER, 2);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_QUIVER_50) {
+        Inventory_ChangeUpgrade(UPG_QUIVER, 3);
+        INV_CONTENT(ITEM_BOW) = ITEM_BOW;
+        AMMO(ITEM_BOW) = CAPACITY(UPG_QUIVER, 3);
+        return ITEM_NONE;
+
+    } else if (item == ITEM_BOMB_BAG_20) {
+        if (CUR_UPG_VALUE(UPG_BOMB_BAG) == 0) {
+            Inventory_ChangeUpgrade(UPG_BOMB_BAG, 1);
+            INV_CONTENT(ITEM_BOMB) = ITEM_BOMB;
+            AMMO(ITEM_BOMB) = CAPACITY(UPG_BOMB_BAG, 1);
+            return ITEM_NONE;
+
+        } else {
+            AMMO(ITEM_BOMB)++;
```

Comment:
Odd, this doesn't look like the others...

Reply:
What were you referring to with this comment?

Reply:
The way that it ++s outside the check, instead of the two things the others do.